types/presto-client/index.d.ts

```diff
@@ -122,9 +122,22 @@ export interface ClientOptions {
     engine?: string;
 }
 
+// Query is a REST call to v1/statements, the `columns` returned is set as the `columns` callback
+// https://github.com/tagomoris/presto-client-node/blob/84f76d981482c5dd710a147ebfe89efa1731d85f/lib/presto-client/index.js#L228
+// Column
+// https://github.com/prestodb/presto/blob/494d5c8f17f1ee19d328535cbfa78914923fc177/presto-client/src/main/java/com/facebook/presto/client/Column.java#L43
 export interface Column {
     name: string;
-    type:
+    typeSignature: ClientTypeSignature;
+    type: string;
+}
+
+// ClientTypeSignature
+// https://github.com/prestodb/presto/blob/494d5c8f17f1ee19d328535cbfa78914923fc177/presto-client/src/main/java/com/facebook/presto/client/ClientTypeSignature.java#L63
+export interface ClientTypeSignature {
+    arguments: any[];
```

Comment:
From the linked type, it looks like this could be further typed as `{ kind: any; object: any }[]`, without spending the time here to figure out all the different values.

Reply:
Addressed in 032f37170b89e4e87a1620a6c8cd1b0a8f2f2cbe